### ISSUE TITLE
Protonmail-Bridge/Hydroxide Compatibilty:  list special-use type error 

### ISSUE THREAD

**Issue by iot-resister** [k9mail/k-9#4729]
### Expected behavior
Listing Special Use Mailbox with IMAP should send the correct type per RFC.  Please see related issue:  https://github.com/emersion/go-imap/issues/357

### Actual behavior
Tell us what happens instead
Server response:
```
Error:
3 LIST (SPECIAL-USE) "" "*" 
3 BAD expected a string
```

### Steps to reproduce
1. Install [hydroxide](https://github.com/emersion/hydroxide) remotely or [protonmail bridge](https://protonmail.com/bridge/) using termux 
2. Get bridge password, ip and port 
3. Configure like so
![](https://f000.backblazeb2.com/file/resistr/FJntyJwHGfPHkCQnhOPrXAOA.png)

### Environment
K-9 Mail version:  5.6

Android version: 9

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors) and attach them here:

```
05-05 12:22:02.995 24517 27142 D ImapConnection: Connecting to 127.0.0.1 as /127.0.0.1
05-05 12:22:03.000 24517 27142 V ImapConnection: conn37659324 <<< #null# [OK, [CAPABILITY, IMAP4rev1, LITERAL+, SASL-IR, AUTH=PLAIN], IMAP4rev1 Service Ready]
05-05 12:22:03.001 24517 27142 D ImapConnection: Saving [AUTH=PLAIN, SASL-IR, LITERAL+, IMAP4REV1] capabilities for conn37659324
05-05 12:22:03.002 24517 27142 V ImapConnection: conn37659324>>> 1 AUTHENTICATE PLAIN
05-05 12:22:03.003 24517 27142 V ImapConnection: conn37659324<<<#+# []
05-05 12:22:04.999 24517 27142 V ImapResponseParser: conn37659324<<<#1# [OK, [CAPABILITY, IMAP4rev1, LITERAL+, SASL-IR, SPECIAL-USE, MOVE], AUTHENTICATE completed]
05-05 12:22:05.000 24517 27142 D ImapConnection: Saving [SASL-IR, SPECIAL-USE, LITERAL+, MOVE, IMAP4REV1] capabilities for conn37659324
05-05 12:22:05.002 24517 27142 I ImapConnection: pathPrefix is unset but server does not have NAMESPACE capability
05-05 12:22:05.003 24517 27142 V ImapConnection: conn37659324>>> 2 LIST "" ""
05-05 12:22:05.008 24517 27142 V ImapResponseParser: conn37659324<<<#null# [LIST, [\Noselect], /, /]
05-05 12:22:05.009 24517 27142 V ImapResponseParser: conn37659324<<<#2# [OK, LIST completed]
05-05 12:22:05.010 24517 27142 D ImapConnection: Got path delimiter '/' for conn37659324
05-05 12:22:05.012 24517 27142 D ImapStore: Folder auto-configuration: Using RFC6154/SPECIAL-USE.
05-05 12:22:05.013 24517 27142 V ImapConnection: conn37659324>>> 3 LIST (SPECIAL-USE) "" "*"
05-05 12:22:05.015 24517 27142 V ImapResponseParser: conn37659324<<<#3# [BAD, expected a string]
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin: Error while testing settings
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin: com.fsck.k9.mail.store.imap.NegativeImapResponseException: Command: LIST (SPECIAL-USE) "" "*"; response: #3# [BAD, expected a string]
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.mail.store.imap.ImapResponseParser.readStatusResponse(ImapResponseParser.java:118)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:757)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:743)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.mail.store.imap.ImapStore.autoconfigureFolders(ImapStore.java:237)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.mail.store.imap.ImapStore.checkSettings(ImapStore.java:314)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.activity.setup.AccountSetupCheckSettings$CheckAccountTask.checkIncoming(AccountSetupCheckSettings.java:497)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.activity.setup.AccountSetupCheckSettings$CheckAccountTask.checkServerSettings(AccountSetupCheckSettings.java:467)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.activity.setup.AccountSetupCheckSettings$CheckAccountTask.doInBackground(AccountSetupCheckSettings.java:424)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at com.fsck.k9.activity.setup.AccountSetupCheckSettings$CheckAccountTask.doInBackground(AccountSetupCheckSettings.java:402)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at android.os.AsyncTask$2.call(AsyncTask.java:333)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-05 12:22:05.017 24517 27142 E AccountSetupCheckSettin:       at java.lang.Thread.run(Thread.java:764)
```

**Comment by cketti** [comment]
This is fixed in 5.7xx. See https://github.com/k9mail/k-9/pull/3842

**Comment by iot-resister** [comment]
Fixed in 5.7. Awesome!
